src/library_async.js

```diff
@@ -20,228 +20,6 @@ mergeInto(LibraryManager.library, {
   },
 
 #if ASYNCIFY
```

Comment:
Maybe we can also just move this into an assert at the top of the file, and always conditionally include this in `src/modules.js`?

Reply:
What do we do about the stub functions then?

Reply:
Oh I didn't see those.   I guess we still need it then.

Reply:
Hmm.. can't we just remove those and let them turn into linker errors?

Reply:
Hm, it looks like `src/modules.js` already does this, actually: https://github.com/emscripten-core/emscripten/blob/master/src/modules.js#L136

I guess this means the stub functions are dead code then? Or are they still needed in case someone does  `-lasync.js` or `-s AUTO_JS_LIBRARIES` without `-s ASYNCIFY`?

Reply:
`AUTO_JS_LIBRARIES` is on by default, and only normally disabled as part of `-s STRICT`, so those stubs are still normally available to non `ASYNCIFY` users... but I'm not sure they should be.   Such users should probably just get link error if they reference those functions.   Although I could be missing some reason why a non-ASYNCIFY need those symbols to exist at link time?

Reply:
I have implemented stub removal in a separate commit for now 269a6e956f3921db0e13ebb935e524bc1dcfee90, let's see if any tests fail. The only potential reason I see for keeping those symbols around is that maybe some user code happens to reference them in a dead code path that emscripten can't eliminate for some reason? But I really don't know. Maybe @kripken has something to say about this one.

Reply:
Yeah If we do that it can be followup for sure.  Lets land this change as a small cleanup.

Reply:
Hm, I actually made the commit on this branch, but I guess I can split it off too if you want.